libvast/src/system/index.cpp

```diff
@@ -345,76 +345,86 @@ caf::error index_state::load_from_disk() {
+            // not, which is hard to figure out here.
+            auto partition = self->spawn(passive_partition, uuid, accountant,
+                                         static_cast<store_actor>(global_store),
+                                         filesystem, path);
+            self->request(partition, caf::infinite, atom::erase_v)
+              .then(
+                [uuid](atom::done) {
+                  VAST_DEBUG("index erased partition {} during startup", uuid);
+                },
+                [uuid](const caf::error& e) {
+                  VAST_WARN("index failed to erase partition {} during "
+                            "startup: "
+                            "{}",
+                            uuid, e);
+                });
+          }
+        }
+        for (auto const* id : *transform_v0->output_partitions()) {
+          const auto uuid = vast::uuid::from_flatbuffer(*id);
+          const auto from_partition = fmt::format(
+            VAST_FMT_RUNTIME(transformer_partition_path_template()), uuid);
+          const auto to_partition = partition_path(uuid);
+          const auto from_partition_synopsis = fmt::format(
+            VAST_FMT_RUNTIME(transformer_partition_synopsis_path_template()),
+            uuid);
+          const auto to_partition_synopsis = partition_synopsis_path(uuid);
+          std::filesystem::rename(from_partition, to_partition);
+          std::filesystem::rename(from_partition_synopsis,
+                                  to_partition_synopsis);
         }
       }
-      for (auto const* id : *transform_v0->output_partitions()) {
-        const auto uuid = vast::uuid::from_flatbuffer(*id);
-        const auto from_partition = fmt::format(
-          VAST_FMT_RUNTIME(transformer_partition_path_template()), uuid);
-        const auto to_partition = partition_path(uuid);
-        const auto from_partition_synopsis = fmt::format(
-          VAST_FMT_RUNTIME(transformer_partition_synopsis_path_template()),
-          uuid);
-        const auto to_partition_synopsis = partition_synopsis_path(uuid);
-        std::filesystem::rename(from_partition, to_partition);
-        std::filesystem::rename(from_partition_synopsis, to_partition_synopsis);
-      }
-    }
-    // TODO: This does not handle store files, which may already have been
-    // written. Since a store file may also be written before the partition
-    // itself, there does not currently seem to be a bulletproof way of handling
-    // this.
-    std::filesystem::remove_all(markersdir);
+      // TODO: This does not handle store files, which may already have been
+      // written. Since a store file may also be written before the partition
+      // itself, there does not currently seem to be a bulletproof way of
+      // handling this.
+      std::filesystem::remove_all(markersdir);
+      return caf::none;
+    }();
+    if (error)
+      VAST_ERROR("{} failed to finish leftover transforms: {}", *self, error);
```

Comment:
I think this should be a warning and not an error: After all, this will not stop VAST from functioning.

Reply:
Done.